x-pack/legacy/plugins/security/server/lib/authentication/index.ts

```diff
@@ -4,5 +4,27 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-export { AuthenticationResult } from './authentication_result';
-export { DeauthenticationResult } from './deauthentication_result';
+import { Legacy } from 'kibana';
+import { KibanaRequest } from '../../../../../../../src/core/server';
+import { SecuritySetupContract } from '../../../../../../plugins/security/server';
+
+export function initAuthentication({ getAuthenticatedUser }: SecuritySetupContract) {
+  return {
+    isAuthenticated: async (request: Legacy.Request) => {
```

Comment:
I'm not completely sure how the core function works, does it work in the same way (legacy is based on [this](https://github.com/elastic/kibana/blob/743f631bd690d0c6ad2b14b7e732a96612045357/x-pack/legacy/plugins/security/server/lib/get_user.ts#L14-L18), the fact that we don't get the latest status for every call shouldn't matter I guess)? Technically since Core now knows about authc we should use core function everywhere if possible.